src/infrastructures/AudioRenderer.ts

```diff
@@ -242,32 +207,43 @@ export class Transport implements BaseTransport {
  * 登録されているシーケンスのイベントをスケジュールします。主に保存用途です。
  */
 export class OfflineTransport implements BaseTransport {
-  private sequences: SoundSequence[] = [];
+  private schedulers = new Map<Sequence, EventScheduler>();
+
+  private createScheduler(sequence: Sequence) {
+    let scheduler: EventScheduler | undefined;
+    if (sequence.type === "audio") {
+      scheduler = new AudioEventScheduler(
+        sequence.audioPlayer,
+        sequence.audioEvents
+      );
+    } else {
+      scheduler = new NoteEventScheduler(
+        sequence.instrument,
+        sequence.noteEvents
+      );
+    }
+    return scheduler;
+  }
```

Comment:
別にこちらの実装でも全く問題ないのですが、こうも書けるかもです！
```suggestion
  private createScheduler(sequence: Sequence) {
    if (sequence.type === "audio") {
      return new AudioEventScheduler(
        sequence.audioPlayer,
        sequence.audioEvents
      );
    } else {
      return new NoteEventScheduler(
        sequence.instrument,
        sequence.noteEvents
      );
    }
  }
```